documents/src/pages/elements/calendar.md

```diff
@@ -190,35 +192,258 @@ ef-calendar {
+<div style="display:flex">
+  <ef-calendar fill-cells view="2023-04" lang="de" id="calendar">
+    <div class="holiday" slot="2023-04-07">7</div>
+    <div class="holiday" slot="2023-04-10">10</div>
+    <div class="holiday" slot="2023-05-01">1</div>
+    <div class="holiday" slot="2023-05-18">18</div>
+    <div class="holiday" slot="2023-05-29">29</div>
+  </ef-calendar>
+  <div>
+    <h5>Germany Public Holidays 2023</h5>
+    <h6>April</h6>
+    <p>
+      07: Good Friday<br>
+      10: Easter Monday
+    </p>
+    <h6>May</h6>
+    <p>
+      01: Labour Day<br>
+      18: Ascension Day<br>
+      29: White Monday
+    </p>
   </div>
+</div>
+```
+```css
+ef-calendar {
+  margin-right: 20px;
+  --holiday-background-color: #F5475B;
+  --holiday-cell-color: #fff;
+}
+
+h5, h6 {
+  margin-top: 0px;
+}
+
+ef-calendar .holiday {
+  background-color: var(--holiday-background-color);
+  color: var(--holiday-cell-color);
+  width: 100%;
+  height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+```
+::
+
+For more advanced use cases, you can use `before-cell-render` event to style your slotted cell. The event is fired before each cell is rendered. [Calendar cell model](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/calendar/types.ts) is provided with the event.
+
+The example below listens to `before-cell-render` event to query slot contents and uses state from `cell` to add CSS classes to the slot content.
+
+```html
+<ef-calendar fill-cells view="2023-04" lang="de">
+  <div class="custom-cell" slot="2023-04-04"></div>
+  <div class="custom-cell" slot="2023-04-24"></div>
+  <div class="custom-cell" slot="2023-04-28"></div>
+  <div class="custom-cell" slot="2023-05-16"></div>
+  <div class="custom-cell" slot="2023-05-25"></div>
+  <div class="custom-cell" slot="2023-05-31"></div>
+  <div class="custom-cell" slot="2023-04"></div>
+  <div class="custom-cell" slot="2023-05"></div>
+  <div class="custom-cell" slot="2023"></div>
 </ef-calendar>
 ```
+
+```javascript
+const calendar = document.getElementById('calendar');
+
+calendar.addEventListener('before-cell-render', (event) => {
+  const sourceCalendar = event.target;
+  const { cell } = event.detail;
+  const customCell = sourceCalendar.querySelector(`[slot="${cell.value}"]`);
+
+  // skip style overriding if there is no content for the cell
+  if (!customCell) { return; }
+
+  // use text from component as calendar has built-in locale support
+  // for instance, Mai instead of May in German
+  customCell.textContent = cell.text;
+
+  // modify classes that match to current cell state
+  const keys = ['range', 'selected'];
+  const customCellClass = customCell.classList;
+  for (const key of keys) {
+    cell[key] ? customCellClass.add(key) : customCellClass.remove(key);
+  }
+});
+```
+
+```typescript
+import { BeforeCellRenderEvent } from '@refinitiv-ui/elements/calendar';
+
+const calendar = document.getElementById('calendar');
```

Comment:
If we use `querySelector('ef-calendar')` stead of `getElementById` we would have,

```ts
const calendar: Calendar | null
```

instead of 

```ts
const calendar: HTMLElement | null
```

Also, let's recheck `typescript` example and make sure that there is no type errors

Reply:
code example updated.